### ISSUE TITLE
Asset tolls support

### ISSUE THREAD

**Issue by corviato1** [hans-schmidt/mastering_ravencoin#2]
Just read the doc, lots of good stuff.

What's your take on Tron's response here.

https://github.com/RavencoinFoundation/ravencoinfoundation.github.io/issues/9

**Comment by hans-schmidt** [comment]
I was not aware of your proposal, which is clearly the same as Asset Tolls. Good idea.

I am not a big fan of RSK. It uses the whole Ethereum EVM and therefore suffers all the security weaknesses which come with that kind of over-powered smart contract implementation. 

A Ravencoin-native implementation of Asset Tolls is far better.

**Comment by corviato1** [comment]
Yeah, it's a bit over my head how it would be deployed so I took Tron's words as fact that it would work better as RSK(also not a big fan).  After reading your doc it reminded me of the proposal and sparked the idea that it actually probably would be better to have that on chain and not as a layer two.   Lmk if there is anything I can do to help bring that to main net.  Down to help test code, even contribute to a bounty to get it moving along.

It may be helpful to divide up the doc into individual proposals so each aspect can get worked on more efficiently.
